### PR TITLE
Reformat cmake lists

### PR DESCRIPTION
**Pull request by StanJBrown**
Cleaned up the cmake lists and made it easier to see what is required in each add_executable call

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -46,59 +66,220 @@ generate_messages(DEPENDENCIES std_msgs sensor_msgs)
...
+    ${OpenCV_LIBS}
+    ${GVars3_LIBRARIES}
+    ${G2O_STUFF_LIBRARY}
+    ${G2O_CORE_LIBRARY}
+    ${G2O_SOLVER_CHOLMOD}
+    ${LAPACK_LIBRARIES}
+    ${BLAS_LIBRARIES}
+)
+
+set(POSE_CALIBRATOR_SRC
```
Comment (stevenwaslander) [inline CMakeLists.txt]: why is this after two of the libs?

File: CMakeLists.txt [hunk 2]
```diff
@@ -46,59 +66,220 @@ generate_messages(DEPENDENCIES std_msgs sensor_msgs)
...
+                 dynamic_reconfigure
+                 pcl_ros
+                 message_runtime
+  DEPENDS SuiteSparse
+          CVD
+          OpenGL
+          GVars3
+)
+
+set(MCPTAM_SRC
```
Comment (stevenwaslander) [inline CMakeLists.txt:139]: I fail to see how this makes things easier, other than putting each file on a separate line ( I do like that).  There are still redundant lists of the same .cc files. Can we organize these in topics that are or aren't used in the executables, like gui, cv, calib, mapping, etc?

Comment (StanJBrown) [inline CMakeLists.txt:139]: Yes we can, but to do so will take a bit of time. I will go through it and see what I can do. We should in theory be able to compile every file that does not start with "main" into one library, or several depending on your preference.

I can cancel this request until that is done if you wish or open a new issue.

File: CMakeLists.txt (hunk 1 above)
Comment (StanJBrown) [inline CMakeLists.txt]: Because I missed it. Will fix.

**Comment by stevenwaslander** [comment]
After merging with master, it appears that compilation fails the first time through but works the second.  Something not happening in the right order. @StanJBrown any chance you can take a look?

**Comment by StanJBrown** [comment]
PR has finally passed after adding the add_dependencies line to the mcptam_functions library in CMakeLists. Ready for merge into the master unless anyone has any objections.

**Comment by stevenwaslander** [comment]
Nicely done. :shipit:

**Pull request merged**